### ISSUE TITLE
add(t, t) loses one gradient accumulation on the GPU path

### ISSUE THREAD

**Issue by felladrin** [felladrin/gguf-trainer#48]
Found while reviewing #47, pre-existing and not reachable from the three shipped architectures, so recording it rather than fixing it blind.

`add(a, b)` binds `a`'s gradient buffer to `DA` and `b`'s to `DB`. When the same tensor is passed twice (`add(t, t)`), both bindings resolve to the same buffer, and the backward kernel does two independent read-modify-writes of it:

```
DA[i] = DA[i] + GY[i];
DB[i] = DB[i] + GY[i];
```

Within one dispatch those race, so `t.grad` gains one contribution instead of two. The CPU reference in `src/model/autograd.ts` accumulates twice and is correct, so this is also a GPU/CPU divergence, but `tests/gpu-parity.ts` never builds `add(t, t)` and none of `gemma3`, `llama` or `qwen3` does either.

Worth either fixing (detect `a === b` and dispatch a scale-by-2 instead) or asserting against it, so a future architecture that writes `add(h, h)` fails loudly rather than training on a halved gradient. The same question applies to `mul(t, t)`.

**Comment by felladrin** [comment]
Closed by #54, but as **not a bug**, so the reasoning belongs here rather than only in the PR.

`add(t, t)` and `mul(t, t)` do bind one buffer to two `read_write` slots, and that is fine on both counts.

**The binding is legal.** WebGPU's compatible-usage-list rule grants an explicit exception:

> Each usage in U is `storage`. Multiple such usages are allowed even though they are writable. This is the usage scope storage exception.

So it is not a validation error waiting to fire on a stricter backend, and an implementation may not assume the two bindings are distinct.

**The arithmetic is defined.** These kernels run one invocation per element, so both writes come from the same thread to the same address in program order. There is no cross-thread race for that exception to expose.

Measured against the CPU reference at n = 1, 6, 257 and 5000, for both ops: maxdiff 0.00e+0. `aliasedBinaryOpParity` in `tests/gpu-parity.ts` now pins it, since both properties are things a future kernel change could break.

Worth recording that I briefly "reproduced" this and the reproduction was wrong: `backward()` seeds only the host scalar, so a non-scalar output leaves the device gradient unseeded and both sides compare zeros. `seedGradFromHost` is what the parity harness uses for exactly that. A bad harness reporting a plausible failure nearly got a fix shipped for a non-bug.

The one case the exception does not cover is a mixed list, binding one buffer as `read` and as `read_write` in the same dispatch. Nothing in the tree does that, and the docstring now says so.

Chasing this did turn up a real bug in the same area, fixed in the same PR: both CPU losses computed `-log(p_target + 1e-12)`, which saturated at 27.63 once the target fell ~88 logits behind the row maximum, while both GPU kernels were exact. See lever 23.
